realgud/common/track.el

```diff
@@ -740,6 +760,10 @@ find a location. non-nil if we can find a location.
 	  nil))
       ))
     )
+(defun realgud-set-eval-string-to-buffer-local (command-hash)
+  "Sets the eval string as a buffer local variable from the COMMAND-HASH."
+  (set (make-local-variable 'realgud-eval-string)
+       (replace-regexp-in-string "%s" "" (gethash "eval" command-hash))))
```

Comment:
i thought its very likely that the command-hash is already available somehow but I couldn't find it.  Is there a better way to do this, or is this fine?

Just to be clear about what this is doing, I need to store the eval command so i can tell if a command was an eval in realgud:track-from-region

Reply:
This feels familiar. In the various debuggers I've written I often have a variable "last command" variable that I set. This is whether it was an "eval" or something else. 

So if I understand this correctly, one might do that here and generalize this use. And it might be useful in addressing another long-standing problem and feature request that people have:

Many debuggers like  pdb don't acknowledge success of  setting a breakpoint. Only failure. Because of this, numerous people have suggested that realgud look for the _absense_ of a complaint after setting a breakpoint in order to store information that a new breapoint has been added so that it can update its internal state. 

If I have this correct this same kind of thing might be useful there. Right?